### ISSUE TITLE
"IllegalArgumentException: Service name must not be null or empty" in v4.0.3 with null or default TracingOptions

### ISSUE THREAD

**Issue by Ultranium** [eclipse-vertx/vertx-tracing#24]
### Version

Vert.x 4.0.3
OpenJDK 15.0.2
Gradle 6.7.1

### Context

After updating from Vert.x 4.0.2 to 4.0.3 I discovered that if I have vertx-opentracing in my classpath, a JaegerTracer is created regardless of whether I specify null TracingOptions for a Vertx instance or don't specify TracingOptions at all. In this case, if no JAEGER_SERVICE_NAME is set, JaegerTacer throws `IllegalArgumentException: Service name must not be null or empty`.

Using `TracingOptions().setFactory(VertxTracerFactory.NOOP)` avoids this problem, but if somewhere in the classpath there is a Vertx creation without this option specified, the IllegalArgumentException is thrown.

For example, [there is a Vertx.vertx() call](https://github.com/vert-x3/vertx-micrometer-metrics/blob/a0a9ce46d34c5430287dc78a81e6b9f030ef870f/src/main/java/io/vertx/micrometer/backends/PrometheusBackendRegistry.java#L73) in `io.vertx.micrometer.backends.PrometheusBackendRegistry`.

Basically, now, if you just create an empty application using the https://start.vertx.io/ and then add the `io.vertx:vertx-opentracing` dependency, it fails to run.

### Reproducer

https://github.com/Ultranium/vertx-4-0-3-jaeger-reproducer

**Comment by doctorpangloss** [comment]
I still have this issue, it is cropping up all the time.

**Comment by vietj** [comment]
can you contribute a fix ?

**Comment by Ultranium** [comment]
@vietj Just created a pull request https://github.com/eclipse-vertx/vert.x/pull/4005

It turns out that the actual problem is in the vertx-core package, although the exception is thrown only if the vertx-opentracing exists in classpath.
